Sources/Atoms/Context/AtomContext.swift

```diff
@@ -102,7 +102,26 @@ public protocol AtomContext {
     /// ```
     ///
     /// - Parameter atom: An atom to reset.
-    func reset(_ atom: some Atom)
+    func reset<Node: Atom>(_ atom: Node)
+
+    /// Resets the value associated with the given atom, and then notifies.
+    ///
+    /// This method only accepts atoms that conform to ``Resettable`` protocol.
+    /// This method resets the value for the given atom and then notifies downstream
+    /// atoms and views. Thereafter, if any other atoms or views are watching the atom, a newly
+    /// generated value will be produced.
+    ///
+    /// ```swift
+    /// let context = ...
+    /// print(context.watch(TextAtom())) // Prints "Text"
+    /// context[TextAtom()] = "New text"
+    /// print(context.read(TextAtom())) // Prints "New text"
+    /// context.reset(TextAtom())
+    /// print(context.read(TextAtom())) // Prints "Text"
```

Comment:
I think those example code should be a bit different from the original `reset` but I have no idea of good example 🤔